sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala

```diff
@@ -253,7 +253,9 @@ class CacheManager extends Logging with AdaptiveSparkPlanHelper {
    * Tries to re-cache all the cache entries that refer to the given plan.
    */
   def recacheByPlan(spark: SparkSession, plan: LogicalPlan): Unit = {
-    recacheByCondition(spark, _.plan.exists(_.sameResult(plan)))
+    if (spark.sessionState.conf.getConf(SQLConf.QUERY_RESULT_CACHE_AUTO_REFRESH)) {
```

Comment:
We can avoid the indentation change by an early return above it?
(not sure if that matters)

Reply:
I don't have a strong opinion. It's just one or two lines of code and indentation change is not that bad.

Reply:
@cloud-fan Could you clarify why do you invoke `spark.sessionState.conf` instead of `spark.conf`?